File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprofit=0;
        int cost=prices[0];
        for(int i=0;i<prices.size();i++)
        {

            maxprofit=max(maxprofit,prices[i]-cost);
            cost=min(cost,prices[i]);

        }
        return maxprofit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxi=float("-inf")
        cost=prices[0]
        for i in range(len(prices)):
            maxi=max(maxi,prices[i]-cost)
            cost=min(cost,prices[i])
        return maxi
